### ISSUE TITLE
关于序列推荐的一些问题

### ISSUE THREAD

**Issue by DX12321** [RUCAIBox/RecBole#1763]
您好，最近在跑序列推荐模型（基于**LightSANs模型**）的过程中，遇到一些问题想请教一下：
Q1. 数据集预处理，想问一下LightSANs论文中对ml-1m数据集是怎么预处理的呢？我跑出的实验结果和论文相比相差较大。主要是传统推荐模型Pop，BPR，FPMC等。
Q2. 评价指标中，实验跑出的Hit评价指标和论文上的相差很大，但实验结果中Recall值和论文中Hit值相差不大，是不是在实验中使用Recall计算，论文中使用Hit作为评价指标呢？

实验超参数是按照LightSANs论文中给出的源码来复现的。

非常感谢！！！

**Comment by zyx1017** [comment]
@peteryang1031  @DX12321 
大佬，对不起，对于LightSANs我有个疑问：
在计算多头注意力得分时，为什么不需要像以前的做法一样，需要把之后的交互给mask掉？
比如以往的做法都会加上一个mask矩阵：attention_scores = attention_scores + attention_mask
而本代码见如下，并没有进行mask操作。对此感到疑问！希望能得到解答！！！
attention_scores = torch.matmul(query_layer, key_layer.transpose(-1, -2))
attention_scores = attention_scores / math.sqrt(self.attention_head_size)

# normalize the attention scores to probabilities.
attention_probs = nn.Softmax(dim=-2)(attention_scores)
attention_probs = self.attn_dropout(attention_probs)
context_layer_item = torch.matmul(attention_probs, value_layer)
